source/module_io/cube_io.h

```diff
@@ -8,45 +8,83 @@
 
 namespace ModuleIO
 {
-bool read_cube(
+extern bool read_cube(
```

Comment:
could you explain why you add extern here?

Reply:
Writing `extern` explicitly will ask the compiler to report an error if the functions in `.h` and `.cpp` are different.